Paper: MDAnalysis: A Python Package for the Rapid Analysis of Molecular Dynamics Simulations

orbeckst opened the pull request.

dschult commented:
Hello, I am a reviewer for this article.  It is my first Github review, so let me know if anything is not clear or if there is a more helpful way to give feedback.

This is a really good presentation of your software. It presents the overall structure of the package as well as connections with other packages. It uses examples throughout in very helpful ways -- this gives the reader a chance to understand the intent as well as the usage of the package. The maturity of the project is helpful in this regard relative to new packages. Your package structure is well organized and leads to easy explanations of functionality.  Your evidence of recent improvements is impressive.

I have a few very minor items which I will create "Line Notes" for. The only more major suggestion I have (which you can choose the depth with which to pursue) is as follows. At the end of your introduction you mention a few other packages for molecular simulations. It would be helpful to very briefly state what the focus of each of those packages is, and their overlap or relationship to the functionality that MDAnalysis provides.  You do this for pytraj, but not for LOOS or mdtraj.

Thanks for your careful document and for your software package. :)

dschult commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -0,0 +1,526 @@
+MDAnalysis currently supports more than 25 different file formats and covers the vast majority of data formats that are used in the biomolecular simulation community, including the formats required and produced by the most popular packages such as NAMD :cite:`Phillips:2005ek`, Amber :cite:`Case:2005uq`, Gromacs :cite:`Abraham:2015aa`, CHARMM :cite:`Brooks:2009pt`, LAMMPS :cite:`Plimpton:1995rw`,  ...
+
+Since the original publication :cite:`Michaud-Agrawal:2011fu`, improvements in speed and data structures make it now possible to work with terabyte-sized trajectories containing up to ~10 million particles.
+MDAnalysis also comes with specialized analysis classes in the ``MDAnalysis.analysis`` module that are unique to MDAnalysis such as *LeafletFinder*, a graph-based algorithm for the analysis of lipid bilayers :cite:`Michaud-Agrawal:2011fu`, or *Path Similarity Analysis* for the quantitative comparison of macromolecular conformational changes :cite:`Seyler:2015fk`.
```
Perhaps it would be helpful to put (```Leaflet``) and (``psa``) after the names *LeafletFinder* and *Path Similarity Analysis*.

orbeckst replied:
done

dschult commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -0,0 +1,526 @@
+   for ts in u.trajectory[::10]:
+       ag.positions
+
+In some cases, such as selecting a specific frames (as in the calculation of time correlation functions), skipping of frames (as in the iterator ``u.trajectory[5000::1000]``), or parallelization over trajectory blocks in a map/reduce pattern :cite:`Tu:2008dq`, sequential reading of trajectories is highly inefficient when the underlying trajectory reader does not implement random access to time fr ...
```
This sentence was hard to read and/or keep track of its flow.  Perhaps split it into two sentences with the examples separated from the main idea.   Also "a specific frames" should either be singular (no s) or plural (no a).

orbeckst replied:
Fixed. Many thanks, this sentence was really hard to parse.

dschult commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -0,0 +1,526 @@
+Most other tools hand the user analysis algorithms as black boxes.
+We want to avoid that and allow the user to adapt an analysis to their needs.
+
+The new ``Contacts`` class is a good example a generic API that allows easy adaptations of algorithms while still offering an easy setup for standard analysis types.
```
example   ---->  example of

orbeckst replied:
fixed

dschult commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -0,0 +1,526 @@
+For native contacts two major types of metrics are considered: ones based on differentiable functions :cite:`Best2013` and ones based on hard cut-offs  :cite:`Franklin2007` (which we set as the default implementation).
+We have designed the API to choose between the two metrics and pass user defined functions to develop new metrics or measure other quantities.
+This generic interface allowed us to implement a "q1q2" analysis :cite:`Franklin2007` on top of the ``Contacts`` class.
+Below is incomplete code example that shows how to implement a q1q2 analysis, the default value for the *method* kwarg is overwritten with a user defined method *radius_cut_q*.
```
is incomplete   -->   is an incomplete

orbeckst replied:
fixed.

Also added half a sentence of context for q1 q2.

dschult commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -0,0 +1,526 @@
+-------------------------
+
+MDAnalysis has been used extensively to study extremely large simulation systems for long simulation times.
+Marrink and co-workers :cite:`Ingolfsson2014` used MDAnalysis to analyze a realistic model of the plasma membrane with 63 different lipid species and over half a million particles for 40 µs. They discovered that transient domains with liquid-ordered character formed and disappeared on the microsecond time scale, with different lipid species clustering in a lipid-specific manner.
```
I think "the plasma membrane" should be "a plasma membrane" because you haven't talked about a membrane yet so the reader doesn't know which membrane "the" refers to. While "a" is general and removes the expectation of knowledge of which membrane you refer to.

orbeckst replied:
In this particular instance, *the* is correct because "[plasma membrane](https://en.wikipedia.org/wiki/Cell_membrane)" (or "cell membrane") is a technical term.

However, I take your point that it is confusing and so I tried to remove some jargon and replaced it with "to analyze a realistic model of the membrane of a mammalian cell with 63 different lipid species".

orbeckst commented:
Hi @dschult,

Many thanks for reviewing our paper and your careful reading. We are going to address your comments over the next week. Your general comments and inline notes are very clear, well considered, and I like that they are clearly "actionable". 

If there are any questions we will just post them on the PR. 

Best wishes,
Oliver

--
Oliver Beckstein
email: orbeckst@gmail.com


Am Jun 20, 2016 um 16:15 schrieb Dan Schult <notifications@github.com>:

> Hello, I am a reviewer for this article. It is my first Github review, so let me know if anything is not clear or if there is a more helpful way to give feedback.

orbeckst commented:
@dschult – I addressed your in-line comments but please do let me know if I missed something.

Still to do: more details on alternative packages as requested in https://github.com/scipy-conference/scipy_proceedings/pull/212#issuecomment-227256135.

suchow commented:
Here's some advice from Proselint (http://github.com/amperser/Proselint), a Python-based linter for prose:

```
176:30: weasel_words.very Substitute 'damn' every time you're inclined to write 'very;' your editor will delete it and the writing will be just as it should be. Found 5 times elsewhere.
```

orbeckst commented:
@dschult – I believe with the latest commits I addressed all your comments. In particular, 726b63a gives more details on the newer packages and I also made clearer what the key ideas in the older packages were to which MDAnalysis is indebted.

I also addressed @suchow 's / proselint's comments in e9e91de.

Please let me know if anything else needs changing or if you believe that the manuscript is ready.

Thank you very much for your feedback — it helped to improve the paper!

dschult commented:
I accept this paper to be merged

orbeckst commented:
Thanks @dschult – I look forward to meeting you next week!

sarostru commented:
Hans Peter Langtangen corresponded with me directly with some feedback, for personal reasons he wasn't able to devote much time, but the fact that he devoted any time I find rather heroic.  He said:

> The article addresses an important application area, reads very well, and brings up new perspectives also of interest for general scientific Python programmers.
> You need a more expert review on the details to lift the article to a higher level - it all reads and looks like an article the way it stands now.

Based on Hans comments, my interpretation is that this paper satisfies the review criteria, but Hans does think it could be better with some expert input.

I won't speak for Hans, but based on that feedback I accept this paper for merge.

sarostru commented:
Due to the complication of the proceedings deadline lining up with everyone's travel schedule to attend the conference we've decided to allow authors to spend an additional 2 days, that is until July 12th to complete any major changes to the submission.  This may impact your enjoyment of the conference, so we hope not too many of you need to take the time.  For reviewers, we ask that you read over this final document before the conference is over and post your decision here in the comments on or before July 17th.

Thanks and have a great Scipy!
Your proceedings co-chairs Scott Rostrup (@sarostru) and Sebastian Benthall (@sbenthall)

orbeckst commented:
@sarostru thanks for relating Hans' comments. I am not entirely sure what I should be doing specifically to "lift the article to a higher level". But if you can let me know what else I need to do in order to get the "ready" tag I'll try to do it.

Btw, SciPy2016 is awesome and being able to present the work was great and led to many new conversations and ideas.

sarostru commented:
Not really sure about "lifting to a higher level", I was just relating that to you.

Next up is preparing for the final camera-ready version, so please go ahead and carefully reread, looking for any formatting or language errors, and @sbenthall and I will help with updating the proceedings build from draft to final.

orbeckst commented:
Ok, will do. Thanks!
